### PR TITLE
:seedling: Bump dependencies, primarily CoreDNS for Kubernetes v1.23

### PR DESCRIPTION
**Pull request by randomvariable**
Signed-off-by: Naadir Jeewa <jeewan@vmware.com>

**What this PR does / why we need it**:
Kubernetes v1.23 will ship with CoreDNS 1.8.6, which means a new version of corefile-migration as per https://github.com/kubernetes/kubernetes/pull/106091 . Whilst I was here, bump all the other dependencies.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by sbueringer** [comment]
Not sure if we want to keep the k/k dependencies sync with the one our CR version is using (even on the patch-level).

File: go.mod
Comment (fabriziopandini) [inline go.mod:28]: FYI then PR for etcd 3.5.1 is not yet merged in k/k, not sure we want to bump before this

File: go.mod
Comment (fabriziopandini) [inline go.mod:37]: Usually we are bumping this in sync with controller runtime, but no strong opinions for a patch version bump

Comment (randomvariable) [reply go.mod:37]: Oh, ok.

File: go.mod
Comment (randomvariable) [reply go.mod:28]: Has it not, it's definitely going in though, so maybe i'll just put a 
/hold

**Comment by randomvariable** [comment]
/unhold

I've removed the etcd bump as it's still baking in https://github.com/kubernetes/kubernetes/pull/105706

File: go.mod
Comment (vincepri) [inline go.mod:22]: These should match controller runtime

Comment (sbueringer) [reply go.mod:22]: Just out of curiosity, why? Do we have dependencies or is it just a matter of consistency?

Comment (sbueringer) [reply go.mod:22]: Good point about pulling multiple versions

Comment (randomvariable) [reply go.mod:22]: Sure, no probs.

Comment (sbueringer) [reply go.mod:22]: @randomvariable PR has been merged

**Review by vincepri — CHANGES_REQUESTED** [review]
@randomvariable Can you rebase?

**Comment by sbueringer** [comment]
I'll take it up in #5761 (co-authored the commit)

So we can close this PR once #5761 has been merged.

**Pull request closed without merging**
